### PR TITLE
Introducing important option

### PR DESCRIPTION
**Pull request by andreasonny83**
- introducing `important` option
- code refactoring
- ESLint updated
- documentation updated

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -4,12 +4,14 @@
...
-    "object-curly-spacing": ["error", "never"]
+    "no-var": 0,
+    "no-param-reassign": 0,
+    "vars-on-top": 0,
+    "prefer-arrow-callback": 0,
+    "prefer-template": 0,
+    "comma-dangle": 0,
+    "func-names": 0,
+    "object-curly-spacing": ["error", "never"],
+    "space-before-function-paren": [1, {"anonymous": "never", "named": "never"}]
```
Comment (princed) [inline .eslintrc:15]: Actually I like the style with space.

File: .eslintrc [hunk 2]
```diff
@@ -4,12 +4,14 @@
...
   },
   "rules": {
-    "no-var": ["off"],
-    "no-param-reassign": ["off"],
-    "vars-on-top": ["off"],
-    "prefer-arrow-callback": ["off"],
-    "prefer-template": ["off"],
-    "comma-dangle": ["error", "never"],
-    "object-curly-spacing": ["error", "never"]
+    "no-var": 0,
```
Comment (princed) [inline .eslintrc:7]: I've come the conclusion that strings are more explicit than number in this case. But of course we don't need arrays.

File: index.js [hunk 3]
```diff
@@ -4,18 +4,21 @@ module.exports = postcss.plugin('postcss-banner', function configure(opts) {
...
-    var comment = text;
-
-    if (!opts.inline) {
-      comment = text.split('\n')
-        .join('\n * ')
-        .concat('\n ');
+    var comment = String(value);
+    var bang = opts.important ? '!' : '';
+
+    if (!!opts.inline) {
```
Comment (princed) [inline index.js:10]: Double bang is redundant

File: README.md [hunk 4]
```diff
@@ -90,10 +89,27 @@ end of the css file.
 
 Type: `Boolean`
 
-Render the banner all in one line
+Default: `false`
+
+Render the banner all in one line.
+
+NOTE: If the string contains new line chars, the resulting comment will
```
Comment (princed) [inline README.md:96]: Do we really need this feature?

File: test/test.js [hunk 5]
```diff
@@ -5,72 +5,98 @@ var expect = require('chai').expect;
...
+    it('should convert values to string', function() {
+      testProcess(input, '/* undefined */\na{ decl: value; }\n/* undefined */',
+        {
+          banner: undefined,
+          footer: undefined,
+          inline: true
+        });
+    });
+
+    it('should add inline bang decorator', function() {
```
Comment (princed) [inline test/test.js:89]: It would be more consistent to add footer to this and next tests.

File: index.js (hunk 3 above)
Comment (ghost) [inline index.js:10]: well it is actually good practice in order to make sure we get back a boolean value. `!!` applied to a value will negate the variable first and then negate again, this way wa always have `true/false` back

File: .eslintrc (hunk 1 above)
Comment (ghost) [inline .eslintrc:15]: feel free to change this as you prefer.

**Comment by ghost** [comment]
Hi, @princed. Feel free to make whatever change you like before merging this feature in your repository.
My PR is just a suggestion in order to improve the plugin functionality

File: README.md (hunk 4 above)
Comment (ghost) [inline README.md:96]: To render the banner all in one line? I don't see anything wrong with that. As we already have that in the source code I suggest to leave that in place. I can think about real case scenarios where people wants to render the banner in multiline while on development mode but they want to render everything in one line on production

File: index.js (hunk 3 above)
Comment (princed) [inline index.js:10]: `if` actually behaves the same way internally

File: README.md (hunk 4 above)
Comment (princed) [inline README.md:96]: I mean, there is no way to render it multiline if you want

Comment (andreasonny83) [inline README.md:96]: I'm not following you here, sorry. If `inline` is not set to true, the banner will be rendered in multiline

**Comment by princed** [comment]
Thanks again!
Published as 3.0.0

**Pull request merged**